src/cpp/shared/artifacts.cpp

```diff
@@ -70,6 +70,7 @@ unsigned char gArtifactLevel[NUM_SUPPORTED_ARTIFACTS] = { 0 };
 std::vector<std::string> names;
 std::vector<std::string> descriptions;
 std::vector<std::string> events;
+std::vector<char> isCursed;
```

Comment:
Why is this a char?

Reply:
I mean....why is it not vector<boolean> ?

Reply:
Sorry, it looks like github munged what I wrote.  I meant that `vector<bool>` is commonly avoided and `vector<char>` is a useful alternative.  I'm using `char` as a 1-byte integer in this case. 
 `vector<int>` would work too but takes up more space.

Reply:
http://www.codingstandard.com/rule/17-1-1-do-not-use-stdvector/

I see. Add a comment about that, and then merge this in yourself.

Reply:
My next change already has changes to this file so I'll add it there.